### PR TITLE
Default storage and extensions reworking

### LINKED ISSUE #1288: Provide default storage system
**Is your feature request related to a problem? Please describe.**
There is no default storage out of the box in the k0s. 
The idea is to allow users create pvc and use them in pods right out of the box.
**Describe the solution you'd like**
The current plan is to use OpenEBS lite mode with an ability to override those settings and easily disable/enable.
https://openebs.io/docs/user-guides/localpv-hostpath#install

**Describe alternatives you've considered**
Writing own provisioner using https://github.com/kubernetes-sigs/sig-storage-lib-external-provisioner requires a lot of effort and doesn't give clear advantages.
https://github.com/rancher/local-path-provisioner uses that provisioner and also doesn't give a lot of advantages over openebs.
**Additional context**

### PR DESCRIPTION
**Pull request by soider**
**Issue**
Fixes #1288 

**What this PR Includes**
- rework helm addons reconciler, make it more clean and easy to maintain
- migrate mentioned reconciler to the controller-runtime
- modify configuration to allow include openebs helm addon based on the boolean flag without manually modifying the helm settings

### CODE REVIEW

**Review by mviitane — CHANGES_REQUESTED** [review]
The storage documentation needs some rework:
https://docs.k0sproject.io/main/storage/

The page could start with explaining that k0s comes with preconfigured OpenEBS for local storage. It can be enabled by... and something about the storage folder configuration and what else is important.

After that we can continue with the existing content explaining:
k0s also supports any other Kubernetes storage solutions with CSI...

**Review by s0j — APPROVED** [review]

**Comment by soider** [comment]
@mviitane added docs and test case as an example

**Review by s0j — APPROVED** [review]

File: docs/storage.md [hunk 1]
```diff
@@ -1,4 +1,94 @@
...
+Use following configuration as an example:
+
+```yaml
+spec:
+  extensions:
+    storage:
+      type: openebs_local_storage
+```
+
+The cluster wil have two storage classes available for you to use:
```
Comment (mviitane) [inline docs/storage.md:16]: wil->will

File: docs/storage.md [hunk 2]
```diff
@@ -1,4 +1,94 @@
...
+
+```shell
+bash-5.1# k0s kc get sc
```
Comment (mviitane) [inline docs/storage.md:19]: The long format would be easier to understand in the example:
`k0s kubectl get storageclass`

File: docs/storage.md [hunk 3]
```diff
@@ -1,4 +1,94 @@
...
+NAME               PROVISIONER        RECLAIMPOLICY   VOLUMEBINDINGMODE      ALLOWVOLUMEEXPANSION   AGE
+openebs-device     openebs.io/local   Delete          WaitForFirstConsumer   false                  24s
+openebs-hostpath   openebs.io/local   Delete          WaitForFirstConsumer   false                  24s
+```
+
+The `openebs-hostpath` is the storage class that maps to the `
```
Comment (mviitane) [inline docs/storage.md:25]: The sentence ends unexpectedly...

File: docs/storage.md [hunk 4]
```diff
@@ -1,4 +1,94 @@
...
+NAME                    READY   STATUS    RESTARTS   AGE
+nginx-d95bcb7db-gzsdt   1/1     Running   0          30s
+bash-5.1# k0s kc get pv
+NAME                                       CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS   CLAIM               STORAGECLASS       REASON   AGE
+pvc-9a7fae2d-eb03-42c3-aaa9-1a807d5df12f   5Gi        RWO            Delete           Bound    default/nginx-pvc   openebs-hostpath            30s
+```
+
+## CSI
+
+K0s also supports any other Kubernetes storage solutions with CSI.
```
Comment (mviitane) [inline docs/storage.md:91]: This sentence should be combined with the following paragraph. Now it's too disconnected, but describing the same thing.

**Pull request merged**